components/heap/heap_trace_standalone.c

```diff
@@ -115,20 +115,55 @@ esp_err_t heap_trace_get(size_t index, heap_trace_record_t *record)
     return result;
 }
 
+// forward define
+static void heap_trace_dump_base(bool internalRam, bool spiRam);
+
+void heap_trace_dump_internal_ram_only(void)
+{
+    heap_trace_dump_base(true, false);
+}
+
+void heap_trace_dump_spi_ram_only(void)
+{
+    heap_trace_dump_base(false, true);
+}
 
 void heap_trace_dump(void)
+{
+    heap_trace_dump_base(true, true);
+}
+
+static void heap_trace_dump_base(bool internalRam, bool spiRam);
 {
     size_t delta_size = 0;
     size_t delta_allocs = 0;
-    printf("%u allocations trace (%u entry buffer)\n",
-           count, total_records);
     size_t start_count = count;
+
+    printf("====== Heap Trace: %u records (%u capacity) ======\n",
+        count, total_records);
+
     for (int i = 0; i < count; i++) {
+
         heap_trace_record_t *rec = &buffer[i];
 
-        if (rec->address != NULL) {
-            printf("%d bytes (@ %p) allocated CPU %d ccount 0x%08x caller ",
-                   rec->size, rec->address, rec->ccount & 1, rec->ccount & ~3);
+        bool shouldPrint = rec->address != NULL &&
+            ((spiRam && internalRam) ||
+             (internalRam && esp_ptr_internal(rec->address)) ||
+             (spiRam && esp_ptr_external_ram(rec->address)));
+
+        if (shouldPrint) {
+
+            const char* label = "";
+            if (esp_ptr_internal(rec->address)) {
+                label = ", Internal-RAM";
+            }
+            if (esp_ptr_external_ram(rec->address)) {
+                label = ", SPI-RAM";
+            }
+
+            printf("%d bytes (@ %p%s) allocated CPU %d ccount 0x%08x caller ",
```

Comment:
It looks like the `%p` and `%s` may be reversed in this print statement.

Reply:
good catch.

Reply:
updated.